Next 16 / Turbopack ビルドで /pom-md, /pom-vscode 配下のドキュメントページが 404 になる

Status: Ready

## 概要

Next.js 16 へのアップグレード (commit a74a240) 以降、`https://pom.pptx.app/pom-md` および `https://pom.pptx.app/pom-vscode` 配下のドキュメントページがすべて 404 を返す状態になっている。`apps/website/content/` 直下のファイル symlink (`api-reference.md` など) は正常に解決されるが、ディレクトリ symlink (`content/pom-md → ../../../packages/pom-md/docs`, `content/pom-vscode → ../../../packages/pom-vscode/docs`) を Turbopack が辿れていない。

## 再現手順

1. `apps/website/` で Next 16.2.4 を install (現状の `package.json` は `"next": "^16"`)。
2. `pnpm build` を実行する。
3. ビルドログに以下のエラーが出力される:

   ```
   [nextra] Error while loading { pathSegments: [ 'pom-md' ] } Error: Cannot find module 'private-next-content-dir/pom-md/index.md'
   [nextra] Error while loading { pathSegments: [ 'pom-vscode' ] } Error: Cannot find module 'private-next-content-dir/pom-vscode/index.md'
   ...
   ```

4. `.next/prerender-manifest.json` を確認すると `/pom-md`, `/pom-md/markdown-syntax`, `/pom-md/pomxml-code-fence`, `/pom-vscode`, `/pom-vscode/configuration`, `/pom-vscode/supported-formats` のルートが含まれていない。
5. 本番 (Vercel) でも該当 URL が 404 (`x-nextjs-prerender: 1`、`x-vercel-cache: PRERENDER`) を返す。

## 期待する挙動

- `/pom-md`, `/pom-md/markdown-syntax`, `/pom-md/pomxml-code-fence` が 200 で表示される。
- `/pom-vscode`, `/pom-vscode/configuration`, `/pom-vscode/supported-formats` が 200 で表示される。
- ランディング/Docs サイドバーから遷移できる。

## 実際の挙動

- 上記すべてのパスが 404 を返す。
- Nextra の `importPage` がディレクトリ symlink 配下の `.md` を `private-next-content-dir/...` 仮想モジュールとして解決できず、`MODULE_NOT_FOUND` で失敗している。
- `generateStaticParams` は fast-glob (default `followSymbolicLinks: true`) でこれらを列挙できているため static params 自体は生成されるが、ページ側の module 解決で落ちて結果的に prerender 結果から除外される。

## 調査メモ

- Next 15.5.14 ではローカルビルド時に `/pom-md/*`, `/pom-vscode/*` がすべて prerender される (確認済み)。
- Next 16.2.4 (現在の lockfile) では Turbopack が `private-next-content-dir/` の解決時にディレクトリ symlink を辿らないため上記エラーが発生する。
- 直下のファイル symlink (`content/api-reference.md` → `packages/pom/docs/api-reference.md` 等) は Turbopack でも問題なく解決される。差分はディレクトリ symlink か否か。
- 上流に類似の挙動を踏んだ issue があれば対応を待てるが、回避策として「symlink ではなく実ディレクトリにする」「`packages/*/docs/` 側を `apps/website/content/` 配下へ移動 (または逆)」「ビルド前に rsync / cp でコピーする」「Webpack ビルドへフォールバック」あたりが候補。

## 受け入れ条件

- [ ] `apps/website/` を Next 16 (`^16`) でビルドして `pom-md`, `pom-vscode` 配下の全ドキュメントページが prerender-manifest に含まれる。
- [ ] 本番 (`https://pom.pptx.app/pom-md`, `https://pom.pptx.app/pom-md/markdown-syntax`, `https://pom.pptx.app/pom-md/pomxml-code-fence`, `https://pom.pptx.app/pom-vscode`, `https://pom.pptx.app/pom-vscode/configuration`, `https://pom.pptx.app/pom-vscode/supported-formats`) が 200 を返す。
- [ ] サイトマップ (`/sitemap.xml`) のエントリと実際のルートが一致する。
- [ ] ビルドログに `[nextra] Error while loading ...` が出ない。

## スコープ外

- `packages/pom-md/`, `packages/pom-vscode/` のドキュメント内容自体の更新。
- Next 15 への revert (Next 16 を維持しつつ解消する方針)。

## 参考

- Next 16 アップグレード commit: a74a240 ("fix: Next.js 16 / Turbopack ビルド失敗を解消")
- `apps/website/content/` 構成変更 commit: fade486 ("docs: pom-md / pom-vscode のドキュメントを Website に追加")
- 関連: https://github.com/vercel/next.js/issues/85238 (Turbopack の動的 require 解析、`next.config.mjs` に既出)

--- a/.claude/rules/pom-md.md
+++ b/.claude/rules/pom-md.md
@@ -18,3 +18,13 @@ pnpm --filter @hirokisakabe/pom-md run test:run     # Run tests
 ```
 
 Pipeline: `Markdown → parseMd() → pom XML string → buildPptx() (core)`
+
+### Documentation Symlinks
+
+`packages/pom-md/docs/` 配下にドキュメントファイルを追加・リネーム・削除した場合、`apps/website/content/pom-md/` 側にもファイル単位の symlink を追加・更新する。Next 16 / Turbopack はディレクトリ symlink を辿れないため、ファイルごとに symlink を張る方針を採用している。
+
+```bash
+# 新規ファイルを追加した場合の例
+cd apps/website/content/pom-md
+ln -s ../../../../packages/pom-md/docs/<new-file>.md <new-file>.md
+```

--- a/.claude/rules/pom-vscode.md
+++ b/.claude/rules/pom-vscode.md
@@ -24,6 +24,16 @@ Pipeline:
 
 To test locally: open `packages/pom-vscode` in VS Code and press F5 to launch Extension Development Host.
 
+### Documentation Symlinks
+
+`packages/pom-vscode/docs/` 配下にドキュメントファイルを追加・リネーム・削除した場合、`apps/website/content/pom-vscode/` 側にもファイル単位の symlink を追加・更新する。Next 16 / Turbopack はディレクトリ symlink を辿れないため、ファイルごとに symlink を張る方針を採用している。
+
+```bash
+# 新規ファイルを追加した場合の例
+cd apps/website/content/pom-vscode
+ln -s ../../../../packages/pom-vscode/docs/<new-file>.md <new-file>.md
+```
+
 ### Release Flow
 
 pom-vscode uses Changesets for versioning (`privatePackages` config). The release is handled by the unified `release.yml` workflow.

--- a/apps/website/content/pom-md
+++ b/apps/website/content/pom-md
@@ -1,1 +1,0 @@
-../../../packages/pom-md/docs

--- a/apps/website/content/pom-md/_meta.ts
+++ b/apps/website/content/pom-md/_meta.ts
@@ -1,0 +1,1 @@
+../../../../packages/pom-md/docs/_meta.ts

--- a/apps/website/content/pom-md/index.md
+++ b/apps/website/content/pom-md/index.md
@@ -1,0 +1,1 @@
+../../../../packages/pom-md/docs/index.md

--- a/apps/website/content/pom-md/markdown-syntax.md
+++ b/apps/website/content/pom-md/markdown-syntax.md
@@ -1,0 +1,1 @@
+../../../../packages/pom-md/docs/markdown-syntax.md

--- a/apps/website/content/pom-md/pomxml-code-fence.md
+++ b/apps/website/content/pom-md/pomxml-code-fence.md
@@ -1,0 +1,1 @@
+../../../../packages/pom-md/docs/pomxml-code-fence.md

--- a/apps/website/content/pom-vscode
+++ b/apps/website/content/pom-vscode
@@ -1,1 +1,0 @@
-../../../packages/pom-vscode/docs

--- a/apps/website/content/pom-vscode/_meta.ts
+++ b/apps/website/content/pom-vscode/_meta.ts
@@ -1,0 +1,1 @@
+../../../../packages/pom-vscode/docs/_meta.ts

--- a/apps/website/content/pom-vscode/configuration.md
+++ b/apps/website/content/pom-vscode/configuration.md
@@ -1,0 +1,1 @@
+../../../../packages/pom-vscode/docs/configuration.md

--- a/apps/website/content/pom-vscode/index.md
+++ b/apps/website/content/pom-vscode/index.md
@@ -1,0 +1,1 @@
+../../../../packages/pom-vscode/docs/index.md

--- a/apps/website/content/pom-vscode/supported-formats.md
+++ b/apps/website/content/pom-vscode/supported-formats.md
@@ -1,0 +1,1 @@
+../../../../packages/pom-vscode/docs/supported-formats.md

